nanodbc/nanodbc.cpp

```diff
@@ -410,8 +410,13 @@ recent_error(SQLHANDLE handle, SQLSMALLINT handle_type, long& native, std::strin
     } while (rc != SQL_NO_DATA);
 
     convert(std::move(result), rvalue);
-    if (size(sql_state) > 0)
-        state = std::string(&sql_state[0], &sql_state[size(sql_state) - 1]);
+    if (size(sql_state) > 0) {
+        state.clear();
```

Comment:
First, thank you very much for your contribution.

Now, would it be reasonable to call `state.reserve(size(sql_state) - 1)` before `push_back`-ing?

Reply:
Sure, though I don't think in practice it'll make any difference, given that `size(sql_state) - 1` is below the Small String Optimization size.